overlord/ifacestate/handlers.go

```diff
@@ -895,6 +904,40 @@ func waitChainSearch(startT, searchT *state.Task) bool {
 	return false
 }
 
+// createConnectTasksForSetupProfiles creates connect tasks and interface hooks for conns and sets their wait chain with regard to the setupProfiles task.
+// The tasks are chained so that:
+//  - prepare-plug-, prepare-slot- and connect tasks are all executed before setup-profiles
+//  - connect-plug-, connect-slot- are all executed after setup-profiles.
+// The "delay-setup-profiles" flag is set on the connect tasks to indicate that doConnect handler should not set security backends up because this will be
+// done later by the setup-profiles task.
+func createConnectTasksForSetupProfiles(st *state.State, setupProfiles *state.Task, conns map[string]*interfaces.ConnRef, autoconnect bool) (*state.TaskSet, error) {
```

Comment:
this is a strange function name,  the goal is to make connect tasks (the separated setup-profile is an optimization), I'm also not super sure why it's passed in instead of created in this helper itself?

"batchConnectTasks" perhaps?

Reply:
Indeed, batchConnectTasks is a better name, renamed.